pkg/nodeconfig/nodeconfig.go

```diff
@@ -87,7 +86,7 @@ func discoverKubeAPIServerEndpoint() (string, error) {
 // NewNodeConfig creates a new instance of nodeConfig to be used by the caller.
 // hostName having a value will result in the VM's hostname being changed to the given value.
 func NewNodeConfig(clientset *kubernetes.Clientset, ipAddress, hostname, clusterServiceCIDR,
-	vxlanPort string, signer ssh.Signer, platform oconfig.PlatformType) (*nodeConfig, error) {
+	vxlanPort, username string, signer ssh.Signer) (*nodeConfig, error) {
```

Comment:
Every commit is changing the signature of `NewNodeConfig`. Not sure if this is a case of too many commits and they all should be squashed with a clear explanation in the commit message 😃